File: 프로그래머스/unrated/181839. 주사위 게임 1/주사위 게임 1.py
def solution(a, b):
    def isOdd(num):
        return num % 2 == 1;
    
    if isOdd(a) & isOdd(b):
        return a**2 + b**2;
    elif isOdd(a) | isOdd(b):
        return 2 * (a + b);
    else:
        return abs(a - b)

File: 프로그래머스/unrated/181839. 주사위 게임 1/주사위 게임 1.js
function solution(a, b) {
    // if(a % 2 === 1 && b % 2 === 1) return a**2 + b**2
    // else if(a % 2 === 1 || b % 2 === 1) return 2 * (a + b)
    // else return Math.abs(a - b)
    
    const isOdd = (num) => {
        return num % 2 === 1;
    }
    
    return isOdd(a) && isOdd(b) ? a**2 + b**2 : isOdd(a) || isOdd(b) ? 2 * (a + b) : Math.abs(a - b);
}